### PR TITLE
GitHub Issue NOAA-EMC/GSI#206. Bug fix which cause HWRF self-cycled DA system failure.

### PR DESCRIPTION
**Pull request by lbi2018**
GSI bug fix to resolve this issue which caused HWRF self-cycled DA failure.

### CODE REVIEW

**Comment by ShunLiu-NOAA** [comment]
The update looks reasonable and the bug is fixed.

**Review by ShunLiu-NOAA — APPROVED** [review]

File: src/gsi/read_l2bufr_mod.f90 [hunk 1]
```diff
@@ -580,6 +580,7 @@ subroutine radar_bufr_read_all(npe,mype)
 		nobs_badsr=nobs_badsr+1
 		cycle
 	     end if
+             if (.not.radar_box) then
```
Comment (MichaelLueken-NOAA) [inline src/gsi/read_l2bufr_mod.f90:583]: Please properly indent code from line 583 through line 692.  There is no indentation from line 583 through 602.  The else on line 602 corresponds with if (.not.radar_box) then, so please align with line 583.  Please indent lines 603 through 692 following the realignment of line 602.  The end if  on line 692 corresponds with the else on line 602 and the if statement on line 583.  Please align the end if with the else and if statements.

**Review by MichaelLueken-NOAA — COMMENTED** [review]
Hi Li,

The regression tests were run and all successfully passed.  The code was compiled in debug mode and the debug tests ran through to completion.

While visually reviewing the changes, the code from lines 583 through 602 aren't indented.  Please indent these lines.  Additionally, the else on line 602 corresponds with the if statement on line 583, so please align the else on line 602 with the if statement on line 583.  Please properly indent the lines from 603 through 691, following the realignment of the else statement.  Finally, the end if on line 692 corresponds with the else on line 602 and the if statement on line 583.  Please ensure the end if is properly aligned.

Once the indentation changes are made, please use the following to push the change back to the PR:

1) git add read_l2bufr_mod.f90
2) git commit --amend
3) Please save and close the commit message without changing it.
4) git push origin master --force

If you have any questions, please let me know.

File: src/gsi/read_l2bufr_mod.f90 (hunk 1 above)
Comment (lbi2018) [reply src/gsi/read_l2bufr_mod.f90]: Made changes to the code and pushed changes back to the PR.

File: src/gsi/read_l2bufr_mod.f90 [hunk 2]
```diff
@@ -580,106 +580,114 @@ subroutine radar_bufr_read_all(npe,mype)
...
-       if (.not.radar_box) then 
-	        iloc=nrbin*(nazbin*(ielbin-1)+(iazbin-1))+irbin
-	        bins(1,iloc,krad)=bins(1,iloc,krad)+range
-	        bins(2,iloc,krad)=bins(2,iloc,krad)+stn_az
-	        bins(3,iloc,krad)=bins(3,iloc,krad)+stn_el
-	        bins(4,iloc,krad)=bins(4,iloc,krad)+rwnd(2,i)
-	        bins(5,iloc,krad)=bins(5,iloc,krad)+rwnd(2,i)**2
-	        bins(6,iloc,krad)=bins(6,iloc,krad)+t
-	        ibins(iloc,krad)=ibins(iloc,krad)+1
+       if (.not.radar_box) then
```
Comment (MichaelLueken-NOAA) [inline src/gsi/read_l2bufr_mod.f90:583]: Line 583 needs to align with line 582, then indent through line 692.  Please see 

/gpfs/dell2/emc/modeling/noscrub/Michael.Lueken/src/gsi/read_l2bufr_mod.f90

for what the change should look like

**Review by MichaelLueken-NOAA — COMMENTED** [review]
Hi Li,

Line 583 needs to align with line 582, then indent through line 692.  Please see the following on WCOSS to see what the change should look like:

/gpfs/dell2/emc/modeling/noscrub/Michael.Lueken/src/gsi/read_l2bufr_mod.f90

File: src/gsi/read_l2bufr_mod.f90 (hunk 2 above)
Comment (lbi2018) [reply src/gsi/read_l2bufr_mod.f90]: Committed changes back to the PR.

**Comment by MichaelLueken-NOAA** [comment]
The due date for the review committee has past with no feedback.  I will now give this update final approval and merge it to the authoritative repository.

**Pull request merged**